### PR TITLE
PWN-8450 - Striga API. Create User

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8450

## Description of Work

Create user API method

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/striga/user/api/response/StrigaCreateUserResponse.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.google.gson.annotations.SerializedName
+
+class StrigaCreateUserResponse(
+    @SerializedName("userId")
+    val userId: String,
+    @SerializedName("email")
+    val email: String,
+    @SerializedName("KYC")
+    val kycDetails: KycDataResponse
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/user/api/response/StrigaCreateUserResponse.kt:11]: Can check for reuse GetStrigaUserResponse?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/striga/user/api/response/StrigaCreateUserResponse.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/striga/user/api/response/StrigaCreateUserResponse.kt:11]: I've checked. After creation, kyc contains only status, if documenation telling truth

**Review by 1sakov — APPROVED** [review]
lgtm

please add unit tests for repositories please

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt [hunk 2]
```diff
@@ -28,6 +35,60 @@ class StrigaUserRepositoryMapper {
...
+                countryCode = map.getOrThrow(StrigaSignupDataType.PHONE_NUMBER)
+            ),
+            dateOfBirth = StrigaCreateUserRequest.DateOfBirthRequest(
+                day = birthday[0],
+                month = birthday[1],
+                year = birthday[2]
+            ),
+            address = StrigaCreateUserRequest.AddressRequest(
+                addressLine1 = map.getOrThrow(StrigaSignupDataType.CITY_ADDRESS_LINE),
+                addressLine2 = "",
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt:74]: emptyString()

File: app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt [hunk 3]
```diff
@@ -28,6 +35,60 @@ class StrigaUserRepositoryMapper {
...
+                city = map.getOrThrow(StrigaSignupDataType.CITY),
+                state = map.getOrThrow(StrigaSignupDataType.CITY_STATE),
+                country = map.getOrThrow(StrigaSignupDataType.COUNTRY),
+                postalCode = map.getOrThrow(StrigaSignupDataType.CITY_POSTAL_CODE)
+            ),
+            occupation = map.getOrThrow(StrigaSignupDataType.OCCUPATION),
+            sourceOfFunds = map.getOrThrow(StrigaSignupDataType.SOURCE_OF_FUNDS),
+            // TODO: it doesn't make sense to get it here, we need some external service
+            ipAddress = "",
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt:83]: emptyString()

**Review by Davran331122 — APPROVED** [review]

**Review by Davran331122 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt (hunk 3 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/striga/user/repository/StrigaUserRepositoryMapper.kt]: I've removed this field, as it is not required

**Pull request merged**